src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,119 @@
+
+    private final TaskRepository taskRepository;
+    private final UserRepository userRepository;
+
+    public TaskService(TaskRepository taskRepository, UserRepository userRepository) {
+        this.taskRepository = taskRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<Task> getTasks() {
+        archiveTasks();
+        return taskRepository.findAll();
+    }
+
+    public List<Task> getAllLoggedUserTasks(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUser(loggedUser);
+    }
+
+    public List<Task> getAllLoggedUserTasksOrderByImportance(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserOrderByImportance(loggedUser);
+    }
+
+    public List<Task> getLoggedUserParticularTasks(Principal principal, Importance importance) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserIdaAndImportance(loggedUser.getId(), importance);
+    }
+
+    public List<Task> getUserTasks(Long userId) {
+        archiveTasks();
+        return taskRepository.findALLByUserId(userId);
+    }
+
+    public Task getTaskById(Long taskId) {
+        archiveTasks();
+        Optional<Task> task = taskRepository.findById(taskId);
+        return task.get();
+    }
+
+    public Task addTask(Task task, Principal principal) {
+        task.setImportance(DEFAULT_IMPORTANCE);
+        task.setCreationDate(LocalDateTime.now());
+        task.setExpirationDate(endOfTheDay());
+        task.setUser(userRepository.findByLogin(principal.getName()).get());
+        return taskRepository.save(task);
+    }
+
+    public Task updateTask(Task task, Long id) {
+        Task updatedTask = taskRepository.findById(id).get();
+        updatedTask.setImportance(task.getImportance());
+        updatedTask.setDescription(task.getDescription());
+        updatedTask.setArchived(task.isArchived());
+        updatedTask.setDone(task.isDone());
+        return taskRepository.save(updatedTask);
+    }
+
+    public void deleteTask(Long id) {
+        taskRepository.deleteById(id);
+    }
+
+
+    public List<Task> getTaskByImportance(Importance importance) {
+        return taskRepository.findAllByImportance(importance);
+    }
+
+
+    public LocalDateTime endOfTheDay() {
+        LocalDateTime now = LocalDateTime.now();
+        return LocalDateTime.of(LocalDate.now(), LocalTime.of(23, 59, 59));
+    }
+
+    public void archiveTasks() {
```

Comment:
This is extremely inefficient and you are calling this almost everywhere. Imagine having 1 million tasks and adding 10 new. Your app would have to iterate through 10 million tasks

Reply:
I know, I asked you about it. You said that we can update this status when user will get tasks. I'm understand it's not a good idea, i just need some more time to figure it out.